event-handler/teleport_events_watcher_test.go

```diff
@@ -136,6 +138,142 @@ func TestNext(t *testing.T) {
 	}
 }
 
+// errMockTeleportEventWatcher is Teleport client mock that returns an error after the first SearchUnstructuredEvents
+type errMockTeleportEventWatcher struct {
+	// events is the mock list of events
+	events []events.AuditEvent
+	t      *testing.T
+	called bool
+}
+
+// SearchEvents is mock SearchEvents method which returns events
+func (c *errMockTeleportEventWatcher) SearchEvents(ctx context.Context, fromUTC, toUTC time.Time, namespace string, eventTypes []string, limit int, order types.EventOrder, startKey string) ([]events.AuditEvent, string, error) {
+	e := c.events
+	c.events = make([]events.AuditEvent, 0) // nullify events
+	return e, "test", nil
+}
+
+// StreamSessionEvents returns session events stream
+func (c *errMockTeleportEventWatcher) StreamSessionEvents(ctx context.Context, sessionID string, startIndex int64) (chan events.AuditEvent, chan error) {
+	return nil, nil
+}
+
+// SearchEvents is mock SearchEvents method which returns events
+func (c *errMockTeleportEventWatcher) SearchUnstructuredEvents(ctx context.Context, fromUTC, toUTC time.Time, namespace string, eventTypes []string, limit int, order types.EventOrder, startKey string) ([]*auditlogpb.EventUnstructured, string, error) {
+	if !c.called {
```

Comment:
I would invert this check.
```suggestion
	if c.called {
		return errors.New("...")
	}
```